app/services/fetch_link_card_service.rb

```diff
@@ -87,13 +87,27 @@ def attempt_opengraph(card, url)
     card.type        = :link
     card.title       = meta_property(page, 'og:title') || page.at_xpath('//title')&.content
     card.description = meta_property(page, 'og:description') || meta_property(page, 'description')
-    card.image       = URI.parse(Addressable::URI.parse(meta_property(page, 'og:image')).normalize.to_s) if meta_property(page, 'og:image')
+
+    if meta_property(page, 'og:image')
+      attempt_opengraph_image(card, Addressable::URI.parse(meta_property(page, 'og:image')).normalize)
+    end
 
     return if card.title.blank?
 
     card.save_with_optional_image!
   end
 
+  def attempt_opengraph_image(card, url)
+    response = http_client.get(url.to_s)
+
+    return if response.code != 200 || !PreviewCard::IMAGE_MIME_TYPES.include?(response.mime_type)
+
+    card.image           = StringIO.new(response.to_s)
```

Comment:
It may be necessary. MediaAttachment seems to catch exceptions ([`process_feed_service.rb#L240-L245`](https://github.com/tootsuite/mastodon/blob/390a2a8ab9bc452ca5f4f07ce8b7f67cb71b1f60/app/services/process_feed_service.rb#L240-L245)).